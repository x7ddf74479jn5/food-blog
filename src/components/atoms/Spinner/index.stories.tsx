import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Spinner from ".";

export default {
  component: Spinner,
  title: "Atoms/Spinner",
} as ComponentMeta<typeof Spinner>;

const Template: ComponentStory<typeof Spinner> = (args) => (
  <div className="fixed inset-0 flex h-screen w-full items-center justify-center ">
    <Spinner {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  size: "w-32 h-32",
};
