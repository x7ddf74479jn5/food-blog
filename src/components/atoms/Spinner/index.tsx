const Spinner: React.VFC = () => {
  return (
    <div className="flex fixed inset-0 justify-center items-center w-full h-screen">
      <div className="w-32 h-32 rounded-full border-b-2 border-green-500 animate-spin"></div>
    </div>
  );
};

export default Spinner;
