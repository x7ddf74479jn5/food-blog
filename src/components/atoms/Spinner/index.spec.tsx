import renderer from "react-test-renderer";

import Spinner from ".";

describe("components/atoms/Spinner", () => {
  it("snapshot", () => {
    const tree = renderer.create(<Spinner size="w-32 h-32" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
