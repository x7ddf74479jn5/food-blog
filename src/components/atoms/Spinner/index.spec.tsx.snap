// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Spinner snapshot 1`] = `
<div
  className="w-32 h-32 rounded-full border-b-2 border-green-500 animate-spin"
/>
`;
